### PR TITLE
Add missing build deps for X stack (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
@boegel please review

### CODE REVIEW

**Comment by boegel** [comment]
@wdpypere please make this `WIP` and let us know when it's done?

**Comment by wpoely86** [comment]
@hajgato This might also interest you. This PR has all the changes I needed to build Qt5 on a machine without any system X libraries present. 

In a nutshell: the build deps should also be copied. If you put `libX11` as a dep for a easyconfigs, you  must also copy the build deps of `libX11`.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 34 out of 35 (29 easyconfigs in this PR)
node2633.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b464d0002fe4d7ac679abc6df5c36972 for a full test report.

**Comment by boegel** [comment]
@wpoely86 failed test report?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 34 out of 35 (29 easyconfigs in this PR)
node2439.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a802ed02babf893feda8a118460c81f2 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 19 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/a3ff639c48be79faf97e16a8ea47a1d2 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 35 out of 35 (29 easyconfigs in this PR)
node2417.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b12134c91e9e1e9538a7c636ab0ae38b for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/a8a4ecd1c8d10cb5927985dc6ca0f168 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 20 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/aa638eb2a07859577bc1b026ec9ea115 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 24 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/d2b22015db1c00b2763c14f20b2fca00 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 24 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/cdd87861945e14e7c3b96a7eb0137145 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 21 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/34208d7041115362735fcbd5866b0885 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 36 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/782000fb05decb393bd714c389deb346 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 36 out of 37 (29 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/3df921ba84fac4b700624391e4381a05 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 59 out of 59 (48 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/b7dc1de6fb980efa70f5a63d1a5ad24a for a full test report.

File: easybuild/easyconfigs/g/glproto/glproto-1.4.17-intel-2016a.eb [hunk 1]
```diff
@@ -12,6 +12,8 @@ toolchainopts = {'optarch': True}
 sources = [SOURCE_TAR_GZ]
 source_urls = [XORG_PROTO_SOURCE]
 
+builddependencies = [('xorg-macros', '1.19.0')]
```
Comment (boegel) [inline easybuild/easyconfigs/g/glproto/glproto-1.4.17-intel-2016a.eb]: @wpoely86 also apply this to the `foss/2016a` easyocnfig please?

File: easybuild/easyconfigs/k/kbproto/kbproto-1.0.7-intel-2016a.eb [hunk 2]
```diff
@@ -11,6 +11,8 @@ toolchain = {'name': 'intel', 'version': '2016a'}
 sources = [SOURCE_TAR_GZ]
 source_urls = [XORG_PROTO_SOURCE]
 
+builddependencies = [('xorg-macros', '1.19.0')]
```
Comment (boegel) [inline easybuild/easyconfigs/k/kbproto/kbproto-1.0.7-intel-2016a.eb]: also in `foss/2016a` easyconfig please

File: easybuild/easyconfigs/g/glproto/glproto-1.4.17-intel-2016a.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/g/glproto/glproto-1.4.17-intel-2016a.eb]: preferably, no. It's manually copying. My goal with this PR is to have a solid base to build Mesa 12 with 2016b for both intel and foss.

File: easybuild/easyconfigs/l/libICE/libICE-1.0.9-intel-2016a.eb [hunk 3]
```diff
@@ -17,6 +17,7 @@ dependencies = [
 
 builddependencies = [
     ('xproto', '7.0.28'),
+    ('xorg-macros', '1.19.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/l/libICE/libICE-1.0.9-intel-2016a.eb]: also in foss/2016a easyconfig please

**Comment by boegel** [comment]
@wpoely86 can you please make the changes consistent across toolchains? I know it's a pain, but it'll bite us in the ass later if we don't...

**Comment by wpoely86** [comment]
I'm happy to do that, starting with the 2016b toolchains.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 49 out of 50 (50 easyconfigs in this PR)
node2010.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f92390169000812a81c52612eb2d3ffe for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 50 out of 50 (50 easyconfigs in this PR)
node2439.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b26c3b10e43291c64a8cb35399eb2fa9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2019.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c22410f4c0968906acd442a4bcf2c1d8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2019.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b3af856c8aac43786a36e2102b45d0a0 for a full test report.

**Comment by boegel** [comment]
@wpoely86 do you mind fleshing out the `Qt5-5.7.0-intel-2016a.eb` easyconfig, since it's failing on SL6 but has nothing to do with the rest?

**Pull request merged**